cli/ballerina-launcher/src/main/java/org/ballerinalang/launcher/util/BRunUtil.java

```diff
@@ -221,15 +222,20 @@ protected static void invokePackageInit(CompileResult compileResult, String pack
     }
 
     private static BValue[] spreadToBValueArray(BValue[] response) {
-        if (!(response != null && response.length > 0 && response[0] instanceof BRefValueArray)) {
+        if (!(response != null && response.length > 0 && response[0] instanceof BValueArray) ||
+                (((((BValueArray) response[0]).elementType == BTypes.typeByte) ||
+                        (((BValueArray) response[0]).elementType == BTypes.typeBoolean) ||
+                        (((BValueArray) response[0]).elementType == BTypes.typeString) ||
+                        (((BValueArray) response[0]).elementType == BTypes.typeFloat) ||
+                        (((BValueArray) response[0]).elementType == BTypes.typeInt)))) {
```

Comment:
shall we move `((BValueArray) response[0]).elementType` to a variable and re-use that?

Reply:
Done.